src/PowerShellEditorServices.Protocol/LanguageServer/Folding.cs

```diff
@@ -0,0 +1,73 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol;
+
+namespace Microsoft.PowerShell.EditorServices.Protocol.LanguageServer
+{
+
+    public class FoldingRangeRequest
+    {
+        public static readonly
+            RequestType<FoldingRangeParams, FoldingRange[], object, object> Type =
+            RequestType<FoldingRangeParams, FoldingRange[], object, object>.Create("textDocument/foldingRange");
+    }
+
+    public class FoldingRangeParams
+    {
+        /// <summary>
+        /// The document to request folding ranges for.
+        /// </summary>
+        public TextDocumentIdentifier TextDocument { get; set; }
+    }
+
+    public class FoldingRange
+    {
+        /// <summary>
+        /// The zero-based line number from where the folded range starts.
+        /// </summary>
+        public int startLine { get; set; }
+
+        /// <summary>
+        /// The zero-based character offset from where the folded range starts. If not defined, defaults to the length of the start line.
+        /// </summary>
+        public int startCharacter { get; set; }
+
+        /// <summary>
+        /// The zero-based line number where the folded range ends.
+        /// </summary>
+        public int endLine { get; set; }
+
+        /// <summary>
+        /// The zero-based character offset before the folded range ends. If not defined, defaults to the length of the end line.
+        /// </summary>
+        public int endCharacter { get; set; }
+
+        /// <summary>
+        /// Describes the kind of the folding range such as `comment' or 'region'. The kind
+        /// is used to categorize folding ranges and used by commands like 'Fold all comments'. See
+        /// [FoldingRangeKind](#FoldingRangeKind) for an enumeration of standardized kinds.
+        /// <summary>
+        public string kind { get; set; }
+    }
+
+    public sealed class FoldingRangeKind
```

Comment:
You should be able to define FoldingRangeKind as an enum and then use that eum type.  Here the Diagnostics.cs file in this same folder - specficically the definition for `public enum DiagnosticSeverity`.

Reply:
`DiagnosticSeverity` maps an int.  This is mapping a name to a string, which unfortunately `enum` can't do.

Reply:
Unless I go down the custom attribute route which seemed overkill ...

https://www.codeproject.com/Articles/11130/String-Enumerations-in-C

Reply:
No, not worth it.  That's an odd way for them to specify this field.  Most of the other "enum" style fields are specified as ints in the protocol.  I guess someone found out that in TypeScript you can have string enums and went a little wild.  ;-)

Reply:
Yeah it's actually a little odd.  The vscode folding uses `Int`s but the LSP uses strings.

https://code.visualstudio.com/docs/extensionAPI/vscode-api#FoldingRangeKind 

vs

https://microsoft.github.io/language-server-protocol/specification#textDocument_foldingRange

Reply:
I bypassed the issue.  Turns out I only use it one place, so I can use a simple `const` instead.